components/Tabs/Tabs.tsx

```diff
@@ -8,23 +8,85 @@ import {
   useState,
 } from "react";
 import HeadlessButton from "components/HeadlessButton";
+import { Dropdown } from "components/Dropdown";
 import { VersionWarning } from "layouts/DocsPage";
 import { DocsContext, getScopes } from "layouts/DocsPage/context";
 import styles from "./Tabs.module.css";
 
-const getSelectedLabel = (
-  tabs: React.ReactComponentElement<typeof TabItem>[]
-): string => {
-  const selectedTab = tabs.find(({ props: { selected } }) => selected);
+/**
+ * An example of using this component.
+ * 
+ * - If at least one TabsItem has an inDropdown prop,
+ * then TabsItems without an inDropdown prop will be displayed in all Dropdown options.
+ * 
+ * <Tabs dropdownCaption="Installing Teleport" dropdownSelected="gatsby">
+    <TabItem label="Download" inDropdown="gatsby, js">
+      [Download MacOS .pkg installer](https://goteleport.com/teleport/download?os=mac) (tsh client only, signed) file, double-click to run the Installer.
+    </TabItem>
 
-  return selectedTab ? selectedTab.props.label : tabs[0]?.props.label;
+    <TabItem label="Homebrew" inDropdown="js, kotlin, python" selected>
+      ```code
+      $ brew install teleport
+      ```
+    </TabItem>
+
+    <TabItem label="Terminal" inDropdown="gatsby, python, java">
+      ```code
+      $ curl -O https://get.gravitational.com/teleport-(=teleport.version=).pkg
+      $ sudo installer -pkg teleport-(=teleport.version=).pkg -target / # Installs on Macintosh HD
+      # Password:
+      # installer: Package name is teleport-(=teleport.version=)
+      # installer: Upgrading at base path /
+      # installer: The upgrade was successful.
+      $ which teleport
+      # /usr/local/bin/teleport
+      ```
+    </TabItem>
+
+    <TabItem scope={["oss", "enterprise"]} label="From Source">
+      ```code
+        # Checkout teleport-plugins
+        $ git clone https://github.com/gravitational/teleport-plugins.git
+        $ cd teleport-plugins/access/mattermost
+        $ make
+      ```
+    </TabItem>
+  </Tabs>
+ */
+
+// getting dropdown options from an individual TabItem
+// we analize all TabItems to gather all available Dropdown options
+const getDropdownFromItem = (inDropdown: string = ""): string[] => {
+  return inDropdown.split(",").map((item) => item.trim());
+};
+
+const isInDropdown = (
+  inDropdown: string,
+  dropdownSelected: string
+): boolean => {
+  return getDropdownFromItem(inDropdown).includes(dropdownSelected);
+};
+
+interface DataTab {
+  label: string;
+  isPreSelected: boolean;
+}
+
+interface TabsInDropdowns {
+  [key: string]: DataTab[];
+}
+
+const getSelectedTab = (tabsMeta: DataTab[]) => {
+  const selected = tabsMeta.find((t) => t.isPreSelected);
+  return selected ? selected.label : tabsMeta[0].label;
 };
 
 export interface TabItemProps {
-  selected?: boolean;
-  scope?: string | string[];
   label: string;
   children: React.ReactNode;
+  selected?: boolean;
+  scope?: string | string[];
+  inDropdown?: string;
```

Comment:
Like options in a Select component, I think `inDropdown` should be called `options`:

```ts
  selected?: boolean;
  scope?: string | string[];
  options?: string;
```